src/integrator_interface.jl

```diff
@@ -161,9 +169,16 @@ Calculate next step of `integrator`.
             integrator.integrator.u = integrator.uprev
             integrator.integrator.uprev = integrator.uprev_cache
         end
-        recursivecopy!(integrator.integrator.k, integrator.k_integrator_cache)
+        recursivecopy!(view(integrator.integrator.k, 1:integrator.kshortsize),
+                       integrator.k_integrator_cache)
+
+        # remove additional interpolation steps
+        resize!(integrator.k, integrator.kshortsize)
```

Comment:
I think you might as well keep them if you've already calculated them. The reason why they're lazy is because usually it's a lot more work to calculate those steps and not a memory thing.

Reply:
But even without resizing `integrator.k` is different from the `integrator.integrator.k` that is calculated later in https://github.com/devmotion/DelayDiffEq.jl/blob/4c1e4d1773d565e002e18ac78147034f6d538384/src/integrator_utils.jl#L258. The additional interpolation steps in `integrator.k` are still the interpolation steps of the previous fixed point iteration, so when `resize!` is removed here we would then need an additional `ode_addsteps!`.

Reply:
I see. Makes sense.